waku/v2/protocol/waku_store/waku_store.nim

```diff
@@ -457,9 +457,42 @@ proc query*(w: WakuStore, query: HistoryQuery, handler: QueryHandlerFunc) {.asyn
     return
 
   waku_store_messages.set(response.value.response.messages.len.int64, labelValues = ["retrieved"])
-
   handler(response.value.response)
 
+  
+proc findLastSeen*(list: seq[IndexedWakuMessage]): float = 
+  var lastSeenTime = float64(0)
+  for iwmsg in list.items : 
+    lastSeenTime = if iwmsg.msg.timestamp>lastSeenTime: iwmsg.msg.timestamp else: lastSeenTime 
+  return lastSeenTime
+
+proc resume*(ws: WakuStore){.async, gcsafe.} =
+  ## fetch the message history of the DefaultTopic since the last seen message in the db
+  debug "resume"
+  var currentTime = epochTime()
```

Comment:
Assuming clock is accurate here, so maybe add that to log as well and make a note?

People are definitely going to be bitten by this, so at some point in future we might want to consider requiring NTP server or so to be running, or print some form of warning when starting up a node about clock

Reply:
Clocks can be out of sync for at most 20 sec. 
In more details, for every pair of nodes in the system lets say P and Pi, the following relation holds between their clocks i.e., `ti` and `t` 
`ti = t + ci + fi(t)`

`ci` is  a constant value and represents the skew
`fi(t)` is a function of `t` cnd signifies the clock drift between P and Pi

We assume that `-10s<ci<+10s` and `-10s<fi(t)<+10s`
Considering these, `ti` and `t` can differ by at most `+-20s`

Following the preceding assumption, the offline time window is adjusted by a 20 sec offset (lines 475-478 copied below). Let me know whether this makes sense.  
```
# adjust the time window with an offset of 20 seconds
  let offset: float64 = 200000
  currentTime = currentTime + offset
  lastSeenTime = max(lastSeenTime - offset, 0)
```

Reply:
To clarify, I understand why it is done, how it is done and that it is done. What I suggest is:

- Add a comment in the code describing this assumption / link to spec portion describing it
- Log this as part of debug statement with time so people can debug it with faulty clocks
- Note for future that we'll have to figure out a way to make this user friendly
--  potentially add to README? way to detect/sync NTP as part of setup script? etc (out of scope of PR but something to be mindful of)

Reply:
I see that a comment was added, thanks!

Reply:
For the issue, I'd probably create a small issue in nim-waku to figure out better way to detect/deal with out of sync real time clocks. Then we can figure out when it makes sense to attack it (not prio now, more relevant when people start using this on wonky desktop/server setups without things like NTP).

Reply:
Thanks for the clarification!
Created the issue https://github.com/status-im/nim-waku/issues/547